src/tree/common_row_partitioner.h

```diff
@@ -205,6 +244,14 @@ class CommonRowPartitioner {
         ctx, tree, this->Partitions(), p_out_position,
         [&](size_t idx) -> bool { return gpair[idx].GetHess() - .0f == .0f; });
   }
+
+ private:
```

Comment:
Maybe we should extract these into a different class or struct with its own utility member functions? Also, the `CommonRowPartitioner` has private members, please place them together.